pkg/network/protocols/http/monitor_test.go

```diff
@@ -262,44 +308,89 @@ func TestHTTPMonitorIntegrationWithNAT(t *testing.T) {
-	t.Run("with keep-alives", func(t *testing.T) {
-		testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
-			EnableKeepAlives: true,
-		})
-	})
-	t.Run("without keep-alives", func(t *testing.T) {
-		testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
-			EnableKeepAlives: false,
+
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
+			t.Run("with keep-alives", func(t *testing.T) {
+				testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
+					EnableKeepAlives:   true,
+					EnableTCPTimestamp: &TCPTimestamp.value,
+				})
+			})
+			t.Run("without keep-alives", func(t *testing.T) {
+				testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
+					EnableKeepAlives:   false,
+					EnableTCPTimestamp: &TCPTimestamp.value,
+				})
+			})
 		})
-	})
+	}
 }
 
 func TestUnknownMethodRegression(t *testing.T) {
-	monitor := newHTTPMonitor(t)
-
 	// SetupDNAT sets up a NAT translation from 2.2.2.2 to 1.1.1.1
 	netlink.SetupDNAT(t)
 
-	targetAddr := "2.2.2.2:8080"
-	serverAddr := "1.1.1.1:8080"
-	srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
-		EnableTLS:        false,
-		EnableKeepAlives: true,
-	})
-	defer srvDoneFn()
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
+			monitor := newHTTPMonitor(t)
 
-	requestFn := requestGenerator(t, targetAddr, emptyBody)
-	for i := 0; i < 100; i++ {
-		requestFn()
-	}
+			targetAddr := "2.2.2.2:8080"
+			serverAddr := "1.1.1.1:8080"
+			srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
+				EnableTLS:          false,
+				EnableKeepAlives:   true,
+				EnableTCPTimestamp: &TCPTimestamp.value,
+			})
+			defer srvDoneFn()
 
-	time.Sleep(10 * time.Millisecond)
-	stats := monitor.GetHTTPStats()
+			requestFn := requestGenerator(t, targetAddr, emptyBody)
+			for i := 0; i < 100; i++ {
+				requestFn()
+			}
 
-	for key := range stats {
-		if key.Method == MethodUnknown {
-			t.Error("detected HTTP request with method unknown")
-		}
+			// give time to collect/aggregate
+			time.Sleep(3 * time.Second)
+			stats := monitor.GetHTTPStats()
+			telemetry := monitor.httpTelemetry
+			hits1XX := telemetry.hits1XX.Get()
+			dropped := telemetry.dropped.Get()
+			rejected := telemetry.rejected.Get()
+			malformed := telemetry.malformed.Get()
+
+			requestsSum := 0
+			for key := range stats {
+				if key.Method == MethodUnknown {
+					t.Error("detected HTTP request with method unknown")
+				}
+				// we just want our requests
+				if strings.HasPrefix(key.Path.Content, "/request-") &&
+					key.DstPort == 8080 &&
+					key.DstIPLow == 0x01010101 {
```

Comment:
either keep a comment to explain what is 0x01010101, or use the existing utility of `FromLowHigh` and use a readable address :)

Reply:
done